single_kernel_mongo/utils/helpers.py

```diff
@@ -73,3 +80,29 @@ def wrapper(self, *args, **kwargs):
 
 mongodb_only = partial(charm_kind_only, charm_kind=CharmKind.MONGOD)
 mongos_only = partial(charm_kind_only, charm_kind=CharmKind.MONGOS)
+
+
+def get_ldap_connection_status(state: LdapState) -> StatusBase:
+    """Checks if the LDAP connection is working or not."""
+    bind_dn = state.bind_user
+    bind_password = state.bind_password
+    base_dn = state.base_dn
+
+    if not base_dn:
+        return BlockedStatus("Missing base DN.")
+
+    if not state.ldaps_urls:
+        return BlockedStatus("Missing LDAPS URLs.")
+
+    try:
+        for ldap_uri in state.ldaps_urls:
+            tls = LDAPTls(validate=ssl.CERT_REQUIRED, version=ssl.PROTOCOL_TLSv1_2)
+            server = LDAPServer(host=ldap_uri, use_ssl=True, tls=tls)
+            conn = LDAPConnection(server, user=bind_dn, password=bind_password)
+            conn.bind()  # We consider sufficient to be able to bind.
+            conn.unbind()
```

Comment:
Can you explain this?

Reply:
We ensure that we have successful connectivity with LDAP server using TLS.
It's not mandatory because mongod will refuse to start if configured with ldap but cannot connect to ldap but this can prevent unsuccessful restarts.